src/Acrobot.jl

```diff
@@ -0,0 +1,220 @@
+import Flux.Tracker
+"""
+Copyright 2013, RLPy http://acl.mit.edu/RLPy
+Credit: Alborz Geramifard, Robert H. Klein, Christoph Dann,
+               William Dabney, Jonathan P. How
+License: "BSD 3-Clause"
+Author: "Christoph Dann <cdann@cdann.de>"
+"""
+mutable struct AcrobotEnv
+    viewer
+    observation_space::Box
+    action_space::Discrete
+    state
+    dt::Float32
+    LINK_LENGTH_1::Float32  # [m]
+    LINK_LENGTH_2::Float32  # [m]
+    LINK_MASS_1::Float32  # [kg] mass of link 1
+    LINK_MASS_2::Float32  # [kg] mass of link 2
+    LINK_COM_POS_1::Float32  # [m] position of the center of mass of link 1
+    LINK_COM_POS_2::Float32  # [m] position of the center of mass of link 2
+    LINK_MOI::Float32  # moments of intertia for both links
+    MAX_VEL_1::Float32
+    MAX_VEL_2::Float32
+    AVAIL_TORQUE::Array{Float32, N} where N
+    torque_noise_max::Float32
+    book_or_nips::String
+end
+
+
+function AcrobotEnv()
+    dt = 2f-1
+    LINK_LENGTH_1 = 1f0
+    LINK_LENGTH_2 = 1f0
+    LINK_MASS_1 = 1f0
+    LINK_MASS_2 = 1f0
+    LINK_COM_POS_1 = 5f-1
+    LINK_COM_POS_2 = 5f-1
+    LINK_MOI = 1f0
+
+    MAX_VEL_1 = Float32(4 * pi)
+    MAX_VEL_2 = Float32(9 * pi)
+
+    AVAIL_TORQUE = [-1f0, 0f0, 1f0]
+    torque_noise_max = 0f0
+
+    # use dynamics equations from the nips paper or the book
+    book_or_nips = "book"
+    # action_arrow = nothing
+    # domain_fig = nothing
+    # actions_num = 3
+
+    viewer = nothing
+    high = [1f0, 1f0, 1f0, 1f0, MAX_VEL_1, MAX_VEL_2] #//
+    low = -high
+    observation_space = Box(low, high, Float32)
+    action_space = Discrete(3)
+    state = nothing
+
+    AcrobotEnv(viewer, observation_space, action_space, state, dt, LINK_LENGTH_1,
+               LINK_LENGTH_2, LINK_MASS_1, LINK_MASS_2, LINK_COM_POS_1, LINK_COM_POS_2,
+               LINK_MOI, MAX_VEL_1, MAX_VEL_2, AVAIL_TORQUE, torque_noise_max, book_or_nips)
+end
+
+
+function reset!(env::AcrobotEnv)
+    env.state = param(rand(Float32, 4) * 2f-1 .- 1f-1)
+    return _get_ob(env)
+end
+
+#=
+function custom_reset!(env::AcrobotEnv, args=[-0.04616826f0, -0.04653378f0,  0.07321809f0, -0.04461966f0])
+    env.state = param(args)
+    return _get_ob(env)
+end
+=#
+
+function step!(env::AcrobotEnv, action)
+    s = env.state
+    torque = env.AVAIL_TORQUE[action]
```

Comment:
Is it possible to a function for converting action to torque? so that we can define gradient for this.

Reply:
@tejank10 If I'm understanding you correctly, I think you are suggesting something like `action_to_torque(env::AcrobotEnv, action) = env.AVAIL_TORQUE[action]`, yes? 

Quick question, are we doing this so that we can do `torque = action_to_torque.(action)`, where `action` can now be an Array? So now, instead of calling `for` over each action in, say, `action_list`, we can just end up passing `action_list` to the step func?

Reply:
Unsure about why we'd want to define `gradient` for the action_to_torque function. It only pulls out the torque from a Discrete space based on what the action...

Reply:
Ohh, I missed that action space is discrete here. My bad.